api/models/route.go

```diff
@@ -128,9 +133,52 @@ func (r *Route) Validate() error {
 		return ErrRoutesInvalidMemory
 	}
 
+	_, err = sanitizeCPUs(r.CPUs)
+	return err
+}
+
+func (r *Route) Sanitize() error {
+
+	sanitized, err := sanitizeCPUs(r.CPUs)
+	if err != nil {
+		return err
+	}
+
+	r.CPUs = sanitized
 	return nil
 }
 
+// sanitizeCPUs validates and cleans up CPUs string
+func sanitizeCPUs(CPUs string) (string, error) {
+
+	if CPUs == "" {
+		return "", nil
+	}
+
+	// first check if this is valid: MinCPU < val < MaxCPUs
+	cpu, err := strconv.ParseFloat(CPUs, 64)
+	if err != nil || cpu < MinCPUs || cpu > MaxCPUs {
+		return "", ErrRoutesInvalidCPUs
+	}
+	if cpu == 0.0 {
+		return "", nil
+	}
+
+	// IMPORTANT: This is varchar(16) in datastore sql
+	// Enforce .2 precision and clean up padded zeros, preceding sign, etc.
+	sanitized := fmt.Sprintf("%.2f", cpu)
+
+	// now check if by sanitizing we ended up losing precision
+	// if we lost precision, then an unsupported format (eg. 0.22020)
+	// was submitted.
+	tmp, err := strconv.ParseFloat(sanitized, 64)
+	if err != nil || cpu != tmp {
+		return "", ErrRoutesInvalidCPUs
```

Comment:
can we just floor this? the error message doesn't make it clear that they must specify only up to 2 decimals precision, and it doesn't seem to warrant an error really to lop off any more precision. up to you, but need to make error clearly state what's wrong or floor / ceiling I think.

Reply:
I was thinking about adding this to the documentation where expected format is 2 decimal points precision. We kind of do the same for other settings, eg. we don't have ErrMemoryTooLarge or TimeoutTooBig, so I stuck with one error code. I could round it silently too. Not sure.

Reply:
I think that's fine, just the error could be more specific about its format / bounds so that users could debug it quicker than having to go look up the docs (we'll get a slack message long before then). rounding for this seems much easier and pretty inconsequential.